jax/_src/scipy/special.py

```diff
@@ -914,6 +914,7 @@ def body_fun(i, p_val):
   return p
 
 
+@partial(jit, static_argnums=(0, 1))
```

Comment:
It's better to avoid `jit` with `static_argnums` when defining user-facing functions, becuase this currently causes the function to fail when called with keyword arguments. In this case, I'd probably leave off the `JIT`: the user can always wrap with JIT if they'd like it to be compiled, and since `_gen_associated_legendre` is JIT compiled, this annotation does not add much.

Reply:
Thanks for the explanations! Removed `JIT`. Done.